src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -806,16 +807,32 @@ private static bool BuildTargetFrameworkNode(PackageSpec packageSpec, KeyValuePa
             return true;
         }
 
-        private static void PopulateFrameworkReferences(ISet<string> frameworkReferences, JObject properties, string frameworkReferenceName)
+        private static void PopulateFrameworkReferences(ISet<FrameworkDependency> frameworkReferences, JObject properties, string frameworkReferenceName, string packageSpecPath)
         {
-            var frameworkRefProperty = properties[frameworkReferenceName];
+            var frameworkRefs = properties[frameworkReferenceName] as JObject;
 
-            if (frameworkRefProperty != null)
+            if (frameworkRefs != null)
             {
-                IEnumerable<string> frameworkRefArray;
-                if (TryGetStringEnumerableFromJArray(frameworkRefProperty, out frameworkRefArray))
+                foreach (var frameworkReference in frameworkRefs)
                 {
-                    frameworkReferences.AddRange(frameworkRefArray);
+                    if (string.IsNullOrEmpty(frameworkReference.Key))
+                    {
+                        throw FileFormatException.Create(
+                                   "Unable to resolve frameworkReference ''.",
```

Comment:
Shouldn't this be a localizable resource string?  Also, it looks like there should be a replacement token in there, like `'{0}'`.

Reply:
It's an internal error that improves debugability. 

The FileFormatException knows how to append the value (2nd param) and the spec path (3rd param).

Reply:
Remove:  ` ''`.